src/08_example/02_handling_connections_concurrently.md

```diff
@@ -1,34 +1,38 @@
 # Handling Connections Concurrently
+The problem with our code so far is that `listener.incoming()` is a blocking iterator.
+The executor can't run other futures while `listener` waits on incoming connections,
+and we can't handle a new connection until we're done with the previous one.
 
-The problem with our code so far is that `listener.incoming()` is a blocking iterator;
-we can't read a new request from this stream until we're done with the previous one.
-One strategy to work around this is to spawn a new Task to handle each connection in the background:
-```rust
-{{#include ../../examples/08_03_concurrent_tcp_server/src/main.rs:main_func}}
-```
-
-This works because under the hood, the `async_std` executor runs `handle_connection` on a separate thread.
-However, this doesn't completely solve our problem: `listener.incoming()` still blocks the executor.
-Even if connections are handled in separate threads, futures running on the main thread
-are blocked while `listener` waits on incoming connections.
+In order to fix this, we'll transform `listener.incoming()` from a blocking Iterator
+to a non-blocking Stream. Streams are similar to Iterators, but can be consumed asynchronously.
+For more information, see the [chapter on Streams](../05_streams/01_chapter.md).
 
-In order to fix this, we can replace our blocking `std::net::TcpListener` with the non-blocking `async_std::net::TcpListener`.
+First, let's replace our blocking `std::net::TcpListener` with the non-blocking `async_std::net::TcpListener`,
+and update our connection handler to accept an `async_std::net::TcpStream`:
+```rust,ignore
+{{#include ../../examples/08_04_concurrent_tcp_server/src/main.rs:handle_connection}}
+```
 
-This change prevents `listener.incoming()` from blocking the executor
-by allowing us to `await` the next TCP connection on this port.
-Now, the executor can yield to other futures running on the main thread
+This change provides two benefits.
```

Comment:
nit, slight rephrase: Having a stream provides two benefits:

1. It does not block the executor while awaiting the next item.
2. Streams can optionally be processed concurrently, using the `for_each_concurrent` method. Since we need to support multiple simultaneous requests at the same time, we will take advantage of this feature.

Feel free to rephrase however you like. I want to highlight (a) that streams and concurrent execution are different (streams are useful in non-concurrent cases too) and (b) what advantage concurrency brings to our application (handling multiple requests simultaneously).